com.unity.render-pipelines.core/Editor/MaterialUpgrader.cs

```diff
@@ -299,7 +308,13 @@ static void SaveAssetsAndFreeMemory()
             AssetDatabase.Refresh();
         }
 
-        private static bool ShouldUpgradeShader(Material material, HashSet<string> shaderNamesToIgnore)
+        /// <summary>
+        /// Checking if the passed in value is a path to a Material.
+        /// </summary>
+        /// <param name="material">Material to check.</param>
+        /// <param name="shaderNamesToIgnore">HashSet of strings to ignore.</param>
+        /// <return>Returns true if the passed in material's shader is not in the passed in ignore list.</return>
+        public static bool ShouldUpgradeShader(Material material, HashSet<string> shaderNamesToIgnore)
```

Comment:
This function is so simple I don't even know if we should bother making it public. It's just checking if shaderNamesToIgnore contains material.shader.name plus null check.

Reply:
That is correct, either it is Public, or we duplicate the code. Or we make it internal visible to somehow. Those are the options what I can see.

Reply:
Moved duplicate code into URP